src/storage/txn/scheduler.rs

```diff
@@ -491,16 +496,27 @@ impl<E: Engine, L: LockManager> TxnScheduler<E, L> {
     }
 
     pub(in crate::storage) fn run_cmd(&self, cmd: Command, callback: StorageCallback) {
-        // write flow control
-        if cmd.need_flow_control() && self.inner.too_busy(cmd.ctx().region_id) {
-            SCHED_TOO_BUSY_COUNTER_VEC.get(cmd.tag()).inc();
+        let tag = cmd.tag();
+        let fail_with_busy = |callback: StorageCallback| {
+            SCHED_TOO_BUSY_COUNTER_VEC.get(tag).inc();
             callback.execute(ProcessResult::Failed {
                 err: StorageError::from(StorageErrorInner::SchedTooBusy),
             });
+        };
+        // write flow control
+        if cmd.need_flow_control() && self.inner.too_busy(cmd.ctx().region_id) {
```

Comment:
After introducing the memory quota limit, is the pending bytes/pending commands `too_busy` check still necessay?
@MyonKeminta What do you think?

Reply:
I think memory and write flow are two different dimensions of limits

Reply:
Let decide it later, I need some time for revisiting this flow control and the flow control in `process_write`.